cpu/native/mtdi/native_mtdi.c

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2016  OTA keys S.A.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <stdio.h>
+
+#include "mtdi.h"
+#include "native_mtdi.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+static const char _fname[] = "MEMORY.bin";
+
+#ifndef NATIVE_MTDI_PAGE_SIZE
+#define NATIVE_MTDI_PAGE_SIZE     256
+#endif
+#ifndef NATIVE_MTDI_PAGE_NUM
+#define NATIVE_MTDI_PAGE_NUM      32768
+#endif
+#ifndef NATIVE_MTDI_SECTOR_SIZE
+#define NATIVE_MTDI_SECTOR_SIZE   4096
+#endif
+#ifndef NATIVE_MTDI_SECTOR_NUM
+#define NATIVE_MTDI_SECTOR_NUM    2048
+#endif
+#ifndef NATIVE_MTDI_FLASH_SIZE
+#define NATIVE_MTDI_FLASH_SIZE    (NATIVE_MTDI_SECTOR_SIZE * NATIVE_MTDI_SECTOR_NUM)
+#endif
+
+static mtdi_sta_t _init(mtdi_dev_t *mtdi_dev)
+{
+    (void) mtdi_dev;
+
+    FILE *f = fopen(_fname, "r");
+
+    if (!f) {
+        f = fopen(_fname, "w+");
+        for (unsigned long i = 0; i < NATIVE_MTDI_FLASH_SIZE; i++) {
+            fputc(0xff, f);
+        }
+    }
+
+    fclose(f);
+
+    return MTDI_STA_INIT;
+}
+
+static mtdi_result_t _read(mtdi_dev_t *mtdi_dev, unsigned char *buff, unsigned long addr, unsigned long size)
+{
+    (void) mtdi_dev;
+
+    DEBUG("native_mtdi: read from page %lu count %lu\n", addr, size);
+
+    FILE *f = fopen(_fname, "r");
+    fseek(f, addr, SEEK_SET);
+    fread(buff, 1, size, f);
+    fclose(f);
+
+    return MTDI_RES_OK;
+}
+
+static mtdi_result_t _write(mtdi_dev_t *mtdi_dev, const unsigned char *buff, unsigned long addr, unsigned long size)
+{
+    (void) mtdi_dev;
+
+    DEBUG("native_mtdi: write from page %lu count %lu\n", addr, size);
+
+    FILE *f = fopen(_fname, "r+");
+    fseek(f, addr, SEEK_SET);
+    fwrite(buff, 1, size, f);
```

Comment:
Is this driver meant to emulate a flash? in that case, buff should be AND:ed with the existing data (it's only possible to write zeroes, any existing zeroes will remain)